src/lib/map.js

```diff
@@ -123,3 +123,12 @@ module.exports.getRoomScore = function (toRoom, fromRoom, opts = {}) {
   }
   return scores.WEIGHT_NEUTRAL ? scores.WEIGHT_NEUTRAL : PATH_WEIGHT_NEUTRAL
 }
+
+// Our own implementation of depricated function
+module.exports.isRoomAvailable = function (roomName) {
+  const homeRoomStatus = Game.map.getRoomStatus(_.values(Game.spawns)[0].room.name).status
```

Comment:
This line throws an exception when there are no spawns left. The empire is only dead when all creeps are dead too. I'd check for `Game.creeps` as a fallback when there are no `Game.spawns`.

Reply:
So there's no point in checking spawns then. The check can be done against creeps object as they are in the same zone. This can be replaced with 'normal' constant, but it will hurt people using Novice and Respawn zones.

Reply:
Right after spawning there are no creeps yet though :wink:

Reply:
No guarantee of a spawn either if using `auto` mode on private servers 😉 May want to use `_.keys(Game.rooms)[0]` instead since that will always exist.

Reply:
An observer could have been used to observe a room outside the zone